suse_migration_services/fstab.py

```diff
@@ -113,3 +116,27 @@ def export(self, filename):
 
     def get_devices(self):
         return self.fstab
+
+    def get_root_disk(self):
+        """Get the disk name for / partition."""
+        lsbkl_mountpoint_call = Command.run(
+            ['lsblk', '-r', '-n', '-o', 'NAME,MOUNTPOINT']
+        )
+
+        for entry in lsbkl_mountpoint_call.output.split(os.linesep):
+            block_record = entry.split()
+            if len(block_record) >= 2 and block_record[1] == '/':
+                lsbkl_disk_call = Command.run(
+                    [
+                        'lsbkl', '-n', '-o', 'pkname',
+                        '/dev/{}'.format(block_record[0])
+                    ]
+                )
+                return lsbkl_disk_call.output
```

Comment:
I also think the method can be done simpler. What we want is the unix disk device that has the root partition. From the read we know the unix device that holds the root partition. So it could be written like:

```python
def _get_root_disk(self):
    for fstab_entry in self.fstab:
        if fstab_entry.mountpoint == '/':
           self.root_disk = Command.run(
               ['lsblk', '-no', 'pkname', fstab_entry.unix_device]
           ).output
           break
```

Reply:
I would call _get_root_disk at the end of the read() method and initialize the class with empty self.root_disk